expression/builtin_time.go

```diff
@@ -5865,7 +5865,7 @@ func (c *makeTimeFunctionClass) getFunction(ctx sessionctx.Context, args []Expre
 	tp, flen, decimal := args[2].GetType().EvalType(), 10, 0
 	switch tp {
 	case types.ETInt:
-	case types.ETReal, types.ETDecimal:
+	case types.ETReal, types.ETDecimal, types.ETDuration:
```

Comment:
1. How about ETDatetime and ETTimestamp?
2. For ETDuration, decimal must between 0 and 6, so it won't need to check decimal again.

I have a PR to refine the setting type of duration. You can wait for my PR.

Reply:
Thanks for you review.
According to definnition of the maketime function, third argument is seconds.
https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_maketime

Min value for DATE,DATETIME, TIMESTAMP, YEAR is too large for seconds, maketime function alway get null result if use these type as argument and there are no need to care about decimal.
https://dev.mysql.com/doc/refman/8.0/en/date-and-time-type-syntax.html

Only TIME for datetime syntax is valid for maketime funtion, so I think ETDatetime and ETTimestamp should use default settings for decimals.

Reply:
mysql> SELECT MAKETIME(12,15,date'2001-01-30');
Field   1:  `MAKETIME(12,15,date'2001-01-30')`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       TIME
Collation:  binary (63)
Length:     10
Max_length: 0
Decimals:   0
Flags:      BINARY


+----------------------------------+
| MAKETIME(12,15,date'2001-01-30') |
+----------------------------------+
| NULL                             |
+----------------------------------+
1 row in set (0.00 sec)

Even if the result is `NULL`, the decimal and flen should be set correctly.